python/pyarrow/parquet/__init__.py

```diff
@@ -3116,6 +3136,14 @@ def file_visitor(written_file):
                              "'basename_template' parameter instead. For "
                              "usage see `pyarrow.dataset.write_dataset`"),
             FutureWarning, stacklevel=2)
+    if existing_data_behavior is not None:
+        warnings.warn(
+            _DEPR_MSG.format("existing_data_behavior", " Specify "
+                             "'use_legacy_dataset=False' while constructing "
+                             "the ParquetDataset, and then use the "
+                             "'basename_template' parameter instead. For "
```

Comment:
this last part is not relevant here I think (copy paste from partition_filename_cb?)

Reply:
This could also be an error like the schema/partitioning/use_threads handling above?

Reply:
Sorry I was careless, will correct.